core/domain/story_domain.py

```diff
@@ -656,6 +656,25 @@ def require_valid_title(cls, title):
         if title == '':
             raise utils.ValidationError('Title field should not be empty')
 
+    def get_acquired_skill_ids_for_exp_ids(self, exp_ids):
+        """Returns the acquired skill ids of the nodes having the given
+        exploration ids.
+
+        Args:
+            exp_ids: list(str). The list of IDs of the explorations linked to
+            the story.
+
+        Returns:
+            list(str)|None. The list of acquired skill ids for the
+                explorations or None, if no node is linked to them.
+        """
+        acquired_skill_ids = []
+        for node in self.story_contents.nodes:
+            for exp_id in exp_ids:
+                if node.exploration_id == exp_id:
+                    acquired_skill_ids += node.acquired_skill_ids
```

Comment:
May want to make acquired_skill_ids a set in case we pass in duplicates

Reply:
Done.

Reply:
I changed it back to list again because I think it would be better if we can keep the order of the skills. Using set makes it random. I added a check before adding it to the list. Hope that's okay.

Reply:
Makes sense!